hitch/content.md

```diff
@@ -0,0 +1,63 @@
+# What is Hitch?
+
+[Hitch](https://hitch-tls.org/) is a *libev-based* high performance *SSL/TLS proxy* by [Varnish Software](https://varnish-software.com). It is specifically built to terminate TLS connections at high scale and forwards unencrypted HTTP traffic to Varnish or any other HTTP backend.
+
+# How to use this image
+
+Running a Hitch Docker container can be done by using the following command:
+
+```console
+docker run --name=hitch -p 443:443 hitch:latest
+```
+
+This container will expose port `443`, which is required for HTTPS traffic.
+
+## Configuration file and extra options
+
+Without any argument, the container will run `hitch --config=/etc/hitch/hitch.conf`. You can mount your own configuration file to replace the default one:
+
+```console
+docker run -v /path/to/your/config/file:/etc/hitch/hitch.conf hitch
+```
+
+You can also change the path of the configuration file by setting the `HITCH_CONFIG_FILE` environment variable. You can set it to an empty string to disable the configuration file altogether.
+
+Note that extra arguments can be added to the command line. If the first argument starts with a `-`, the arguments are added to the default command line, otherwise they are treated as a command.
+
+> Our assumption is that your backend, Varnish or other, supports both *HTTP/2* and the *PROXY* protocol.
+
+## Connecting to Varnish
```

Comment:
I realize it's "Hitch by Varnish" but up until this point this reads as a totally generic TLS proxy, so as someone who is admittedly not using Varnish currently, this came as a surprise (and I don't see non-Varnish covered later on either) -- is that intentional?

The Hitch documentation implies even `PROXY` protocol support is optional, but my understanding of `HTTP/2` is that it only supports TLS, so I'm not sure how this could forward to an unencrypted backend properly if it requires `HTTP/2` support?  Is that just referring to the default settings?

Reply:
Looooots to unpack here :-) But you are right, it can be confusing.

There's a strong relationship between `hitch` and `varnish`, that you can notably see in the default configuration files:
- https://github.com/varnishcache/pkg-varnish-cache/blob/master/systemd/varnish.service#L26
- https://github.com/varnish/pkg-hitch/blob/master/systemd/hitch.conf#L18

but the two are still completely independent, you can use `HAProxy` to terminate TSL and use the `PROXY` protocol to contact Varnish, or you can use `hitch` in front on any other TCP server that lacks TLS.

Regarding H/2 mandating TLS, yes and no, there are provisions outside of TLS, but that not necessarily the norm.

I think it would make sense to remove the whole section and tackle the whole `hitch`+`varnish` synergy in a separate repo.

Reply:
@tianon @gquintard In https://github.com/docker-library/docs/pull/1675/commits/0d7fe8213d58cadbe5f06990dcfc20d5ed0396fa I removed references to Varnish and mentioned backend connections in general.